ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/TransactionHelper.java

```diff
@@ -26,16 +26,43 @@
 import javax.transaction.xa.XAResource;
 
 /**
+ * Utility class to make transaction status checking and resources registration easier.
+ *
  * @author <a href="mailto:gytis@redhat.com">Gytis Trikleris</a>
  */
 public interface TransactionHelper {
 
+    /**
+     * Check if transaction is active. If error occurs wrap an original exception with {@link JMSException}.
+     *
+     * @return whether transaction is active or not.
+     * @throws JMSException if error occurred getting transaction status.
```

Comment:
Are the reasons for not being able to get the transaction status known? If so it would be useful to state those reasons here.

Similarly for the other methods below. There seems to be a few methods that can throw a generic JMSException so the text could go into the javadoc for the class instead of each method.